### ISSUE TITLE
Error when trying to install locally missing 'nanopore_barcode_binner/cpp_functions.so'

### ISSUE THREAD

**Issue by danarte** [rrwick/Nanopore-barcode-binner#1]
Here is the error I get when trying to install locally with python 3.5.1:

```
$ python setup.py install --user
running install
running build
running build_py
Cleaning previous compilation: make clean
rm -f nanopore_barcode_binner/src/alignment.o nanopore_barcode_binner/src/barcode_align.o
Compiling Nanopore barcode binner: make -j 8
g++ -std=c++14 -Inanopore_barcode_binner/include -fPIC -O3 -D NDEBUG -Wall -Wextra -pedantic -march=native -c -o nanopore_barcode_binner/src/alignment.o nanopore_barcode_binner/src/alignment.cpp
g++ -std=c++14 -Inanopore_barcode_binner/include -fPIC -O3 -D NDEBUG -Wall -Wextra -pedantic -march=native -c -o nanopore_barcode_binner/src/barcode_align.o nanopore_barcode_binner/src/barcode_align.cpp
cc1plus: error: unrecognized command line option "-std=c++14"
cc1plus: error: unrecognized command line option "-std=c++14"
make: *** [nanopore_barcode_binner/src/barcode_align.o] Error 1
make: *** Waiting for unfinished jobs....
make: *** [nanopore_barcode_binner/src/alignment.o] Error 1
running install_lib
running install_egg_info
running egg_info
writing entry points to nanopore_barcode_binner.egg-info/entry_points.txt
writing dependency_links to nanopore_barcode_binner.egg-info/dependency_links.txt
writing nanopore_barcode_binner.egg-info/PKG-INFO
writing top-level names to nanopore_barcode_binner.egg-info/top_level.txt
reading manifest file 'nanopore_barcode_binner.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'nanopore_barcode_binner.egg-info/SOURCES.txt'
removing '/groups/nshomron/artemd/.local/lib/python3.5/site-packages/nanopore_barcode_binner-0.1.0-py3.5.egg-info' (and everything under it)
Copying nanopore_barcode_binner.egg-info to /groups/nshomron/artemd/.local/lib/python3.5/site-packages/nanopore_barcode_binner-0.1.0-py3.5.egg-info
running install_scripts
Installing nanopore_barcode_binner script to /groups/nshomron/artemd/.local/bin
error: [Errno 2] No such file or directory: 'nanopore_barcode_binner/cpp_functions.so'

```

**Comment by rrwick** [comment]
Hmmm, it looks like your GCC version is too old - it doesn't know what `-std=c++14` means.
I think you'll need GCC 4.9.1 or later to compile and install it.

As a side note, sorry about the lack of documentation on this project! I hadn't really refined this program yet. It if generates much interest, I'll flesh it out and give it a nice README. Or perhaps I'll merge this functionality into [Porechop](https://github.com/rrwick/Porechop). As I mentioned in [this Porechop issue](https://github.com/rrwick/Porechop/issues/2), I'm sort of waiting to see how the recent Nanopore basecalling changes look before I decide whether this is a worthwhile project.

**Comment by danarte** [comment]
Wow thanks. I should know better because a lot of software packages require different GCC libraries and I always try to load different libraries when I get errors related to gcc/c, this time for some reason I didn't try.
Anyway I loaded GCC 5.3.1 and the installation went ok.

Regarding interest in the software, I saw porechop mentioned in the ONT forum and I think it's the most user friendly and MinION friendly barcode software available. Leave aside the fact that the adaptors list is already supplied, but your software is fast, reports well and allows customization specific for ONT reads which most of the other programs does not include. The reason I looked into porechop was because I wanted to separate my reads into barcodes, when I saw that porechop reports for each read to what barcode it matches but it will not separate them I was very disappointed. Though then I found this software and now I'll try to use it.

Personally, I think there is/will be interest in your software because barcoding ONT reads differ so much from other NGS workflows and your software does it very nice.

Thanks you.

**Comment by rrwick** [comment]
I got a few requests for it, so I've implemented native barcode demultiplexing in [Porechop](https://github.com/rrwick/Porechop) now! It works similarly to how this program does it. Actually Porechop's implementation is a bit simpler but should be quite sufficient (I think it was a bit over-engineered here).

Give it a try! I'm going to label this project as deprecated now that Porechop does this stuff. Any future development of barcode binning will happen there, not here.
